sections/errorhandling/failfast.md

```diff
@@ -10,35 +10,72 @@ Defensive programming is an approach to improve software and source code, in ter
 
 ### Code example: validating complex JSON input using ‘Joi’
 
-```javascript
+<details>
+<summary><strong>Javascript</strong></summary>
+<pre lang="javascript">
 var memberSchema = Joi.object().keys({
  password: Joi.string().regex(/^[a-zA-Z0-9]{3,30}$/),
  birthyear: Joi.number().integer().min(1900).max(2013),
  email: Joi.string().email()
 });
-
+<br/>
 function addNewMember(newMember) {
  // assertions come first
  Joi.assert(newMember, memberSchema); //throws if validation fails
  // other logic here
 }
+</pre>
+</details>
+
+<details>
+<summary><strong>Typescript</strong></summary>
+<pre lang="typescript">
+const memberSchema = Joi.object().keys({
+ password: Joi.string().regex(/^[a-zA-Z0-9]{3,30}$/),
+ birthyear: Joi.number().integer().min(1900).max(2013),
+ email: Joi.string().email()
+});
+<br/>
+function addNewMember(newMember: Member) {
+ // assertions come first
+ Joi.assert(newMember, memberSchema); //throws if validation fails
+ // other logic here
+}
+</pre>
+</details>
 
-```
 
 ### Anti-pattern: no validation yields nasty bugs
 
-```javascript
+<details>
+<summary><strong>Javascript</strong></summary>
+<pre lang="javascript">
 // if the discount is positive let's then redirect the user to print his discount coupons
 function redirectToPrintDiscount(httpResponse, member, discount) {
     if (discount != 0) {
         httpResponse.redirect(`/discountPrintView/${member.id}`);
     }
 }
-
+<br/>
 redirectToPrintDiscount(httpResponse, someMember);
 // forgot to pass the parameter discount, why the heck was the user redirected to the discount screen?
+</pre>
+</details>
 
-```
+<details>
+<summary><strong>Typescript</strong></summary>
+<pre lang="typescript">
+// if the discount is positive let's then redirect the user to print his discount coupons
+function redirectToPrintDiscount(httpResponse: Response, member: Member, discount: number) {
+  if (discount != 0) {
+    httpResponse.redirect(`/discountPrintView/${member.id}`);
+  }
+}
+<br/>
+redirectToPrintDiscount(httpResponse, someMember, -12);
+// We passed a negative parameter discount, why the heck was the user redirected to the discount screen?

```

Comment:
I altered the original comment here, since we cannot forget to pass the parameter in typescript

Reply:
@kevynb Yes, good point